zfit_physics/models/pdf_tsallis.py

```diff
@@ -0,0 +1,157 @@
+from typing import Optional
+
+import tensorflow as tf
+import zfit
+import zfit.z.numpy as znp
+from zfit import run, z
+from zfit.core.space import ANY_LOWER, ANY_UPPER, Space
+from zfit.util import ztyping
+
+
+@z.function(wraps="tensor")
+def tsallis_pdf_func(x, m, t, n):
+    """Calculate the Tsallis PDF.
+
+    Args:
+        x: value(s) for which the PDF will be calculated.
+        m: mass of the particle.
+        t: width parameter.
+        n: absolute value of the exponent of the power law.
+
+    Returns:
+        `tf.Tensor`: The calculated PDF values.
+
+    Notes:
+        Based on code from `numba-stats <https://github.com/HDembinski/numba-stats/blob/main/src/numba_stats/tsallis.py>`_.
+    """
+    if run.executing_eagerly():
+        if n <= 2:
+            msg = "n > 2 is required"
+            raise ValueError(msg)
+    elif run.numeric_checks:
+        tf.debugging.assert_greater(n, znp.asarray(2.0), message="n > 2 is required")
+
+    x = z.unstack_x(x)
+    mt = znp.sqrt(znp.square(m) + znp.square(x))
+    nt = n * t
+    c = (n - 1) * (n - 2) / (nt * (nt + (n - 2) * m))
+    return c * x * znp.power(1 + (mt - m) / nt, -n)
+
+
+@z.function(wraps="tensor")
+def tsallis_cdf_func(x, m, t, n):
+    """Calculate the Tsallis CDF.
+
+    Args:
+        x: value(s) for which the CDF will be calculated.
+        m: mass of the particle.
+        t: width parameter.
+        n: absolute value of the exponent of the power law.
+
+    Returns:
+        `tf.Tensor`: The calculated CDF values.
+
+    Notes:
+        Based on code from `numba-stats <https://github.com/HDembinski/numba-stats/blob/main/src/numba_stats/tsallis.py>`_.
+    """
+    if run.executing_eagerly():
+        if n <= 2:
+            msg = "n > 2 is required"
+            raise ValueError(msg)
+    elif run.numeric_checks:
+        tf.debugging.assert_greater(n, znp.asarray(2.0), message="n > 2 is required")
+
+    x = z.unstack_x(x)
+    mt = znp.sqrt(m * m + znp.square(x))
+    nt = n * t
+    return znp.power((mt - m) / nt + 1, 1 - n) * (m + mt - n * (mt + t)) / (m * (n - 2) + nt)
+
+
+def tsallis_integral(limits: ztyping.SpaceType, params: dict, model) -> tf.Tensor:
+    """Calculates the analytic integral of the Tsallis PDF.
+
+    Args:
+        limits: An object with attribute limit1d.
+        params: A hashmap from which the parameters that defines the PDF will be extracted.
+        model: Will be ignored.
+
+    Returns:
+        The calculated integral.
+    """
+    lower, upper = limits._rect_limits_tf
+    m = params["m"]
+    t = params["t"]
+    n = params["n"]
+    lower_cdf = tsallis_cdf_func(x=lower, m=m, t=t, n=n)
+    upper_cdf = tsallis_cdf_func(x=upper, m=m, t=t, n=n)
+    return upper_cdf - lower_cdf
+
+
+class Tsallis(zfit.pdf.BasePDF):
+    _N_OBS = 1
+
+    def __init__(
+        self,
+        m: ztyping.ParamTypeInput,
+        t: ztyping.ParamTypeInput,
+        n: ztyping.ParamTypeInput,
```

Comment:
About the names, shouldn't we use `mu` and `sigma`? I agree that it's not a gaussian, therefore, it has a different meaning.
`m` is also fine, I think, just `t` for width seems a bit uncommon? But it's the `T`? So also seems fine overall, just to have it discussed

Or what was the motivation to go with these names instead of others?

Reply:
The motivation is the parameter names here https://arxiv.org/pdf/1207.4724.pdf in eqs 4 and 5 and also here: https://s3.cern.ch/inspire-prod-files-3/384fb3b69db8a55ed08ef9abf02384df

Reply:
I just don't like capital parameter names so I did `T` - > `t`

Reply:
Yes, makes all sense, agree!